text/0000-make-cargo-install-extensible.md

```diff
@@ -0,0 +1,122 @@
+    [package]
+    name = "the-best-cli-app-ever"
+    version = "0.1.0"
+    author = "Ferris the Crab"
+    metainstall = ["cli-install"]
+    
+    [dependencies]
+    cli-install = "0.1.0"
+  ```
+
+### Examples: CLI Applications
+
+For CLI applications installed with `cargo install`, an `install.rs` file could include instructions to generate manpages or a shell completion file. Alternatively, a CLI application could have a `Cargo.toml` that contains a `metainstall` key that points to a crate dependency such as `cli-install` that contains instructions to generate and install manpages and shell completion files.
+
+### Examples: Plugins
+
+For a crate intended as a plugin for an existing program, an `install.rs` file could install the necessary plugin metadata to allow the program to list, enable, disable, and configure the plugin through a UI.
+
+### New Way to Think About Cargo 
+
+This feature extends the new type of thinking that `build.rs` created. Cargo can be seen now as a customizable build and distribution tool, whose functionality can be extended to better fit the specific type of application being built. Cargo is there to ease both the developer experience (by extending builds) as well as a way to ease the end user experience by customizing the distribution functionality of `cargo install` , eliminating the need for extensive documentation and cutting and pasting boilerplate or config.
+
+This feature avoids putting detailed policies and functionality for specific use cases into `cargo install`; instead, it allows the crates.io ecosystem to support various use cases, and facilitates experimentation outside of `cargo` itself.
+
+### New vs. Existing Users
+
+This feature provides an improved developer experience for both new and existing users.
+
+For **existing users**, this feature allows developers a way to improve end-user friendliness; instructions that were once required in extensive documentation can now be moved into code and automated away, creating a significantly more streamlined installation experience for end users.
+
+Additionally, it gives developers the ability to package customized installation steps, which encourages the development of reusable conventions across the ecosystem. **New users** can easily adopt these conventions to provide a consistent experience and consistent functionality.
+
+## Reference-level explanation
+
+This is the technical portion of the RFC. Explain the design in sufficient detail that:
+
+### Interaction with other features
+
+This feature is quite similar to the existing extension functionality of `build.rs`, and `cargo` can be implemented using the same mechanism.
+
+### Corner Cases
+
+- Installing an application should not run the `install.rs` files of any of it’s dependencies, only of the application itself. 
+- When re-installing a binary package this might run a different `install.rs` than the one that was initially run. This might lead to inconsistent/ undefined behavior.
```

Comment:
The use of the phrase "undefined behavior" should probably be reserved for when referring to UB as in your code not having any semantics. If that is what is intended here, then that is concerning.

Reply:
I don't think that's what is intended here, I think it was meant more as "potentially unexpected behavior"/"surprising behavior". I'm happy to update the language. Thanks for the catch!